examples/environments/auction_simulation/env.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+"""The envs used to simulate an auction."""
+from typing import List
+
+from loguru import logger
+
+from agents import Item, Bidder
+
+from agentscope.environment.env import BasicEnv, EventListener
+from agentscope.environment.event import Event, event_func
+
+
+class Auction(BasicEnv):
+    """The auction env."""
+
+    def __init__(
+        self,
+        name: str = None,
+        listeners: List[EventListener] = None,
+    ) -> None:
+        """Initialize the auction env.
+
+        Args:
+            name (`str`): The name of the Auction.
+            listeners (`List[EventListener]`): The listeners.
+        """
+        super().__init__(
+            name=name,
+            listeners=listeners,
+        )
+        self.cur_item = None
+        self.cur_bid_info = None
+
+    @event_func
+    def start(self, item: Item) -> None:
+        """Start bidding for an item.
+        Args:
+            item (`Item`): The item.
+        """
+        self.cur_item = item
+        self.cur_bid_info = None
+        self._trigger_listener(
+            Event("start", args={"item": item}),
+        )
+
+    @event_func
+    def bid(self, bidder: Bidder, item: Item, bid: int) -> None:
+        """Bid for the auction.
+        Args:
+            bidder (`Bidder`): The bidder agent.
+            item (`Item`): The item.
+            bid (`int`): The bid of the bidder.
+        """
+        if self.cur_bid_info is None and bid >= item.opening_price:
+            self.cur_bid_info = {"bidder": bidder, "bid": bid}
+            self._trigger_listener(
+                Event(
+                    "bid",
+                    args={"bidder": bidder, "item": item, "bid": bid},
+                ),
+            )
+        elif bid > self.cur_bid_info["bid"]:  # type: ignore[index]
+            self.cur_bid_info["bidder"] = bidder  # type: ignore[index]
+            self.cur_bid_info["bid"] = bid  # type: ignore[index]
+            self._trigger_listener(
```

Comment:
A bit redundant, move this outside the `if` block and add an `else` branch that returns directly.